src/zenml/utils/warnings/controller.py

```diff
@@ -0,0 +1,162 @@
+        module_name: str,
+        line_number: int,
+        config: WarningConfig,
+    ) -> str:
+        """Helper method to build the warning message string.
+
+        Args:
+            message: The warning message.
+            module_name: The module that the warning call originated from.
+            line_number: The line number that the warning call originated from.
+            config: The warning configuration.
+
+        Returns: A warning message containing extra fields/info based on warning config.
+        """
+        display = f"[{config.code}]({config.category}) - {message}"
+
+        if config.verbosity == WarningVerbosity.MEDIUM:
+            display = f"{module_name}:{line_number} {display}"
+
+        if config.verbosity == WarningVerbosity.HIGH:
+            display = f"{display}\n{config.description}"
+
+        return display
+
+    def _log(
+        self,
+        warning_code: str,
+        message: str,
+        level: LoggingLevels,
+        **kwargs: dict[str, Any],
+    ) -> None:
+        """Core function for warning handling.
+
+        Args:
+            warning_code: The code of the warning configuration.
+            message: The warning message.
+            level: The level of the warning.
+            **kwargs: Keyword arguments (for formatted messages).
+
+        """
+        warning_config = self._warning_configs.get(warning_code)
+
+        # resolves the module and line number of the warning call.
+        module_name, line_number = self._resolve_call_details()
+
+        if not warning_config:
+            # If no config is available just follow default behavior:
+            logger.warning(f"{module_name}:{line_number} - {message}")
+            return
+
+        if warning_config.is_throttled:
+            if warning_code in self._warning_statistics:
+                # Throttled code has already appeared - skip.
+                return
+
+        display_message = self._get_display_message(
+            message=message,
+            module_name=module_name,
+            line_number=line_number,
+            config=warning_config,
+        )
+
+        self._warning_statistics[warning_code] += 1
+
+        if level == LoggingLevels.INFO:
+            logger.info(display_message.format(**kwargs))
+        else:
+            # Assumes warning level is the default if an invalid option is passed.
+            logger.warning(display_message.format(**kwargs))
+
+    def warn(self, *, warning_code: str, message: str, **kwargs: Any) -> None:
+        """Method to execute warning handling logic with warning log level.
+
+        Args:
+            warning_code: The code of the warning (see WarningCodes enum)
+            message: The message to display.
+            **kwargs: Keyword arguments (for formatted messages).
+        """
+        self._log(warning_code, message, LoggingLevels.WARNING, **kwargs)
+
+    def info(self, *, warning_code: str, message: str, **kwargs: Any) -> None:
```

Comment:
```suggestion
    def info(self, *, warning_code: WarningCodes, message: str, **kwargs: Any) -> None:
```
We can use the enum here, no? Same in other places